### PR TITLE
Allow more configuration through env variables

### PR DESCRIPTION
**Pull request by jayniz**
Allows to set the following variables from the lua plugin via
environment variables:

  - NGO_BLACKLIST
  - NGO_CALLBACK_HOST
  - NGO_CALLBACK_SCHEME
  - NGO_CALLBACK_URI
  - NGO_CLIENT_ID
  - NGO_CLIENT_SECRET
  - NGO_DOMAIN
  - NGO_EMAIL_AS_USER
  - NGO_EXTRA_VALIDITY
  - NGO_SECURE_COOKIES
  - NGO_TOKEN_SECRET
  - NGO_USER
  - NGO_WHITELIST

Also allows to supply a `LOCATIONS` environment variable to replace
the demo `location /`. This way you can use the same image for multiple
simple reverse proxies simply by setting environment variables.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -185,17 +185,29 @@ to the ip of your docker daemon, we use `ngo.lol` here. Make sure to add
 
 Docker image has the following env variables for configuration:
 
-* `PORT` is the port for nginx to listen on, defaults to `80`.
+* `DEBUG`: if set, prints the generated nginx configuation on container start
+* `LOCATIONS` can contain `location` directives that will be injected into the nginx configuration on container start. If not set, a demo page is served under `location / {...}`
```
Comment (bobrik) [inline README.md]: Can you wrap this line at 80 chars?

File: docker/run.sh [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#!/bin/sh -e
+#!/bin/bash
```
Comment (bobrik) [inline docker/run.sh]: What's the problem with `/bin/sh` and why do you want to continue on errors?

File: docker/run.sh [hunk 3]
```diff
@@ -19,13 +19,49 @@ if [ -z "${NGO_TOKEN_SECRET}" ]; then
...
+# Overwrite user supplied locations
+if [ "${LOCATIONS}" ]; then
+  echo "#
+# Generated /tmp/nginx-locations.conf
+#
+${LOCATIONS}" > /tmp/nginx-locations.conf
+fi
+
+# Write config file
+sed "s/%NGO_BLACKLIST%/${BLACKLIST}/"                      -i /etc/nginx/sites-available/default && \
```
Comment (bobrik) [inline docker/run.sh]: `-e` in shebang was needed so you don't have to do `&& \`

**Comment by bobrik** [comment]
I wounder if we should remove `sed` and just use `env` and lua ([example](http://blog.doismellburning.co.uk/environment-variables-in-nginx-config/)).

File: docker/run.sh (hunk 3 above)
Comment (jayniz) [inline docker/run.sh]: cool, will change thanks!

File: README.md (hunk 1 above)
Comment (jayniz) [inline README.md]: You got it!

**Comment by jayniz** [comment]
> I wounder if we should remove sed and just use env and lua (example).

Yeah, especially with support for injecting `location` directives, the chances of sed-breaking special characters is on the rise!

**Comment by jayniz** [comment]
@bobrik implemented your feedback the other day

**Comment by bobrik** [comment]
Thanks!

**Pull request merged**